Rust type mismatch error when compiling F# closure code with Fable 4.0.4

### Repro code

```fsx
let rec closure0 () () : int32 =
    0
let v0 : (unit -> int32) = closure0()
()
```

#### Working 4.0.0 code

https://fable.io/repl/#?code=DYUwLgBATiDGG2AewM4FcYAYIAoCUuBAXBAJYB2YAzAEwQC8AsAFARsSYuiQBu2JONOVKQAtAD4ylWgXoJk6LPhbLmQA&html=Q&css=Q

```rs
#![allow(dead_code,)]
#![allow(non_camel_case_types,)]
#![allow(non_snake_case,)]
#![allow(non_upper_case_globals,)]
#![allow(unreachable_code,)]
#![allow(unused_attributes,)]
#![allow(unused_imports,)]
#![allow(unused_macros,)]
#![allow(unused_parens,)]
#![allow(unused_variables,)]
pub mod Test {
    use super::*;
    use fable_library_rust::Native_::Func0;
    use fable_library_rust::Native_::MutCell;
    use fable_library_rust::Native_::on_startup;
    pub fn closure0(unitVar: (), unitVar_1: ()) -> i32 { 0i32 }
    pub fn v0() -> Func0<i32> {
        static v0: MutCell<Option<Func0<i32>>> = MutCell::new(None);
        v0.get_or_init(|| Func0::new(move || Test::closure0((), ())))
    }
    on_startup!((),);
}
```

4.0.1 also works locally, 4.0.2 onward throwing this.

#### Local 4.0.4 (Ubuntu 20.04.5 LTS Codespaces)

```ps1
dotnet fable `
    "$PSScriptRoot" `
    --optimize `
    --lang rs `
    --extension .rs `
    --outDir "$PSScriptRoot"
cargo build --release
```

```rs
#![allow(dead_code,)]
#![allow(non_camel_case_types,)]
#![allow(non_snake_case,)]
#![allow(non_upper_case_globals,)]
#![allow(unreachable_code,)]
#![allow(unused_attributes,)]
#![allow(unused_imports,)]
#![allow(unused_macros,)]
#![allow(unused_parens,)]
#![allow(unused_variables,)]
pub mod Test {
    use super::*;
    use fable_library_rust::Native_::Func0;
    use fable_library_rust::Native_::Func1;
    use fable_library_rust::Native_::MutCell;
    use fable_library_rust::Native_::on_startup;
    pub fn closure0(unitVar: (), unitVar_1: ()) -> i32 { 0i32 }
    pub fn v0() -> Func0<i32> {
        static v0: MutCell<Option<Func0<i32>>> = MutCell::new(None);
        v0.get_or_init(||
                           Func1::new(move |arg10_0040: ()|
                                          Cli::closure0((), ())))
    }
    on_startup!((),);
}
```

```
error[E0308]: mismatched types
  --> test.rs:22:27
   |
20 |     pub fn v0() -> Func0<i32> {
   |                    ---------- expected `Func0<i32>` because of return type
21 |         static v0: MutCell<Option<Func0<i32>>> = MutCell::new(None);
22 |         v0.get_or_init(|| Func1::new(move |arg10_0040: ()| Cli::closure0((), ())))
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `Func0`, found enum `Func1`
   |
   = note: expected enum `Func0<i32>`
              found enum `Func1<(), i32>`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `cli` due to previous error
```

--- a/src/Fable.Transforms/Rust/Fable2Rust.fs
+++ b/src/Fable.Transforms/Rust/Fable2Rust.fs
@@ -1112,7 +1112,7 @@ module Util =
     let isUnitArg (ident: Fable.Ident) =
         ident.IsCompilerGenerated &&
         ident.Type = Fable.Unit &&
-        ident.Name.StartsWith("unitVar")
+        (ident.DisplayName.StartsWith("unitVar") || ident.DisplayName.Contains("@"))
 
     let discardUnitArg (genArgs: Fable.Type list) (args: Fable.Ident list) =
         match genArgs, args with
@@ -1126,7 +1126,7 @@ module Util =
         match genArgs, args with
         | [Fable.Unit], [arg] -> args // don't drop unit arg when generic arg is unit
         | _, [MaybeCasted(Fable.Value(Fable.UnitConstant, _))] -> []
-        | _, [Fable.IdentExpr ident] when ident.Type = Fable.Unit -> []
+        | _, [Fable.IdentExpr ident] when isUnitArg ident -> []
         | _, args -> args
 
     /// Fable doesn't currently sanitize attached members/fields so we do a simple sanitation here.

--- a/tests/Rust/tests/src/ClosureTests.fs
+++ b/tests/Rust/tests/src/ClosureTests.fs
@@ -237,3 +237,10 @@ let ``Non-tail recursive non-capturing closures work`` () =
 let ``Non-tail recursive capturing closures also work`` () =
     let n = fib_rec_clo 2 30
     n |> equal 832040UL
+
+let rec closure0 () () : int32 = 5
+let v0 : (unit -> int32) = closure0()
+
+[<Fact>]
+let ``Closures with multiple unit arguments work`` () =
+    v0() |> equal 5

